extension/src/test/suite/AutoIndent.test.ts

```diff
@@ -0,0 +1,292 @@
+import * as chai from "chai";
+import * as vscode from "vscode";
+import * as fs from "fs";
+import * as path from "path";
+import {findContainers,getIndentation,isCursorInDoubleQuoteExpr} from "../../format/autoIndent";
+import { ReadonlyDocument } from "../../project/readOnlyDocument";
+let assert = chai.assert;
+const testDir = path.join(__dirname + "/../../../extension/src/test");
+const outputDir = path.join(testDir + "/OutputFile");
+let indentTestFile = "";
+let indentTestDoc: vscode.TextDocument;
+
+fs.mkdir(outputDir, { recursive: true }, (err) => {
+  if (err) {
+    return console.error(err);
+  }
+});
+
+function createFakeTextDcoument() {
+  indentTestFile = path.join(testDir + "/SourceFile/autoIndentTestFile.lsp");
+  try {
+    if (fs.existsSync(indentTestFile)) {
+      indentTestDoc = ReadonlyDocument.open(indentTestFile);
+    }
```

Comment:
it should fail on else

Reply:
Yes, But I have put this file there and don't allow it fail. but anyway I can add some warning message here.